### PR TITLE
Add sv starts/ends_with

### PR DESCRIPTION
**Pull request by cyyever**
Fixes #ISSUE_NUMBER

cc @EikanWang @jgong5 @wenzhe-nrv @sanchitintel

### CODE REVIEW

**Comment by cyyever** [comment]
@r-barnes We can use this PR to make some initial fixes.

File: c10/util/string_view.h [hunk 1]
```diff
@@ -598,6 +598,22 @@ constexpr inline void swap(
 }
 using string_view = basic_string_view<char>;
 
+// NOTE: In C++20, this function should be replaced by str.starts_with
+constexpr bool string_view_starts_with(
+    std::string_view str,
+    std::string_view prefix) noexcept {
+  return (prefix.size() > str.size()) ? false
+                                      : prefix == str.substr(0, prefix.size());
```
Comment (Skylion007) [inline c10/util/string_view.h:606]:
```suggestion
    return str.size() >= prefix.size() && str.substr(0, prefix.size()) == prefix;
```

File: c10/util/string_view.h [hunk 2]
```diff
@@ -598,6 +598,22 @@ constexpr inline void swap(
...
+}
+
+// NOTE: In C++20, this function should be replaced by str.ends_with
+constexpr bool string_view_ends_with(
+    std::string_view str,
+    std::string_view suffix) noexcept {
+  return (suffix.size() > str.size())
+      ? false
+      : suffix == (str.substr(str.size() - suffix.size(), suffix.size()));
```
Comment (Skylion007) [inline c10/util/string_view.h:615]:
```suggestion
    return str.size() >= suffix.size() && str.substr(str.size() - suffix.size()) == suffix;
```
Just apply demorgan's law, no need for a ternary operator here.

**Review by Skylion007 — APPROVED** [review]
Nit: make this change and it looks like fine to me.

**Pull request closed without merging**